CONTRIBUTING.md

```diff
@@ -0,0 +1,316 @@
+and grouped in the following order:
+
+1. Standard libraries
+2. Related third-party libraries
+3. Local libraries
+
+For example,
+
+```lua
+local type = type
+local table = { concat = table.concat, insert = table.insert }
+
+local awful = require"awful"
+
+local helpers = require"vicious.helpers"
+```
+
+### String Quotes
+
+In Lua, single-quoted strings and double-quoted strings are the same,
+so the choice is totally up to you, but please be consistent within a module.
+When a string contains single or double quote characters, however,
+use the other one to avoid backslashes in the string. It improves readability:
+
+```lua
+'"key": "value"'        -- good
+"\"key\": \"value\""    -- no good
+```
+
+It is preferable to add a newline immediately after the opening long bracket:
+
+```lua
+foo = [[
+this is a really,
+really,
+really long text]]
+```
+
+### Naming Conventions
+
+Avoid using the characters `l` (lowercase letter el), `O` (uppercase letter oh),
+or `I` (uppercase letter eye) as single character variable names.
+In some fonts, these characters are indistinguishable from the 1's and 0's.
+
+#### Constants
+
+Constants are usually defined on a module level
+and written in all capital letters with underscores separating words.
+Examples include `MAX_OVERFLOW` and `TOTAL`.
+
+
+#### Function and Variable Names
+
+Function names should be lowercase, with words separated by underscores
+as necessary to improve readability.
+
+Variable names follow the same convention as function names.
+
+When you find it difficult to give descriptive names,
+use the functions and variable anonymously.
+
+### Performance Tips
+
+Vicious is meant to be run as part of the Awesome window manager,
+thus any little overhead may defect the responsiveness of the UI.
+While Lua is famous for its performance, there are a few things
+one can do to make use of all of its power.
+
+**Never** use global variables.  This includes the standard libraries,
+which, again, must be localized before use.  Remember, every widget type
+is to be called repeatedly every few seconds.
+
+Use closures when possible:
+
+* Define constants on the module level.
+* Avoid re-fetching the values that are not not meant to change.
+
+However, declare a variable only when you need it, to avoid declaring it
+without an initial value (and therefore you seldom forget to initialize it).
+Moreover, you shorten the scope of the variable, which increases readability. 
+
+### Copyright Header
+
+Vicious is released under the GNU GNU General Public License
+version 2 or later and each contributor holds the copyright
+on their contributing.  To make this collective control effective,
+each source file must include a notice of the following format
+denoting the name of all authors
+
+```lua
+-- <one line to give the program's name and a brief idea of what it does.>
+-- Copyright (C) <year>  <name of author> <<email that can be use for contact>>
```

Comment:
Absolutely agree! We could also add [SPDX IDs](https://spdx.org/ids) while we're at it.

Reply:
While I don't object it, I don't think SPDX is necessary anyhow.  Distros like Debian already examine the source very carefully for the license of every single source file.